### PR TITLE
tests: Validate warnings in best practice tests

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

File: tests/unit/best_practices_positive.cpp [hunk 1]
```diff
@@ -25,6 +25,8 @@ TEST_F(VkPositiveBestPracticesLayerTest, TestDestroyFreeNullHandles) {
     ASSERT_NO_FATAL_FAILURE(InitState());
     ASSERT_NO_FATAL_FAILURE(InitRenderTarget());
 
+    m_errorMonitor->ExpectSuccess(kErrorBit | kWarningBit);
```
Comment (spencer-lunarg) [inline tests/unit/best_practices_positive.cpp:28]: what we really should do is make this the default for all BestPractices tests

also I think `kPerformanceWarningBit` should be added to this

Comment (ziga-lunarg) [reply tests/unit/best_practices_positive.cpp:28]: The problem I ran into trying to make it default is that InitRenderTarget() already produces warnings. That might need a bit more work

Comment (spencer-lunarg) [reply tests/unit/best_practices_positive.cpp:28]: aww, for now I think this is an improvement, so happy to apply this PR, but long term would like a more default solution

**Review by spencer-lunarg — APPROVED** [review]

File: tests/unit/best_practices_positive.cpp (hunk 1 above)
Comment (ziga-lunarg) [reply tests/unit/best_practices_positive.cpp:28]: I agree

Comment (ziga-lunarg) [reply tests/unit/best_practices_positive.cpp:28]: `kPerformanceWarningBit` also seems tricky, for example when we create a `vkt::Buffer`, we only allocate memory that we need, but then we get a small allocation performance warning

**Pull request merged**